core/src/main/scala/caliban/interop/play/play.scala

```diff
@@ -0,0 +1,45 @@
+package caliban.interop.play
+
+import caliban.introspection.adt.__Type
+import caliban.schema.Step.QueryStep
+import caliban.schema.Types.makeScalar
+import caliban.schema.{ ArgBuilder, PureStep, Schema, Step }
+import caliban.{ InputValue, ResponseValue }
+import play.api.libs.json.{ JsValue, Reads, Writes }
+import zio.ZIO
+import zquery.ZQuery
+
+/**
+ * This class is an implementation of the pattern described in https://blog.7mind.io/no-more-orphans.html
+ * It makes it possible to mark play-json dependency as optional and keep Writes defined in the companion object.
+ */
+private[caliban] trait IsPlayJsonWrites[F[_]]
+private[caliban] object IsPlayJsonWrites {
+  implicit val isPlayJsonWrites: IsPlayJsonWrites[Writes] = null
+}
+
+/**
+ * This class is an implementation of the pattern described in https://blog.7mind.io/no-more-orphans.html
+ * It makes it possible to mark play-json dependency as optional and keep Reads defined in the companion object.
+ */
+private[caliban] trait IsPlayJsonReads[F[_]]
+private[caliban] object IsPlayJsonReads {
+  implicit val isPlayJsonReads: IsPlayJsonReads[Reads] = null
+}
+
+object json {
+  implicit val jsonSchema: Schema[Any, JsValue] = new Schema[Any, JsValue] {
+    private def parse(value: JsValue) =
+      implicitly[Reads[ResponseValue]]
+        .reads(value)
+        .asEither
+        .left
+        .map(errs => new Throwable(s"Couldn't decode json: $errs"))
```

Comment:
Maybe not for this PR, but general question to @ghostdogpr should this be a typed error? Seems like we should surface this in a structured way to the user.

Reply:
This is fine, `QueryStep` expects a `Throwable` and will wrap it into an `ExecutionError`.